lymph/edge.py

```diff
@@ -117,23 +120,23 @@ def comp_bayes_net_prob(self, log: bool = False) -> float:
     def comp_stay_prob(self) -> float:
         """Compute the probability of spread per time-step in the hidden Markov model.
 
-        This function dynamically computes the probability of no spread per time-step i.e. 
+        This function dynamically computes the probability of no spread per time-step i.e.
         that the child node will stay in the same state given the states of its parent nodes,
         and the parameters of the edge.
         """
         if self.end.is_binary:
             if self.end.state == 0:
-                return 1 - self.spread_prob
+                return 1 - self._spread_prob
 
         if self.start.state == 1:
             if self.end.state == 0:
-                return 1 - self.spread_prob * self.micro_mod
+                return 1 - self._spread_prob * self.micro_mod
             elif self.end.state == 1:
                 if self.is_growth:
-                    return 1 - self.spread_prob
+                    return 1 - self._spread_prob
         if self.start.state == 2:
             if self.end.state == 0:
-                return 1 - self.spread_prob
+                return 1 - self._spread_prob

```

Comment:
I think you can keep the public `self.spread_prob` here. It still is a property, so there's no need to use the private attribute.